src/Traits.h

```diff
@@ -872,6 +898,55 @@ struct Traits<std::unordered_map<KeyT, T, HashT, KeyEqualT, AllocatorT>> : publi
     }
 };
 
+
+namespace {
+
+template <typename ArchiveT, typename MatrixT>
+ArchiveT & SerializeMatrix(ArchiveT &ar, MatrixT const &value) {
+    ar.serialize(static_cast<typename MatrixT::Index>(value.rows()));
+    ar.serialize(static_cast<typename MatrixT::Index>(value.cols()));
+
+    for (typename MatrixT::Index rowId = 0; rowId < value.rows(); ++rowId) {
+        for (typename MatrixT::Index colId = 0; colId < value.cols(); ++colId) {
+            Traits<typename MatrixT::Scalar>::serialize(ar, value(rowId, colId));
+        }   
+    }
+    
+    return ar;
+}
+
+template <typename MatrixT, typename ArchiveT>
+MatrixT DeserializeMatrix(ArchiveT &ar) {
+    typename MatrixT::Index                 numRows(ar.template deserialize<typename MatrixT::Index>());
+    typename MatrixT::Index                 numCols(ar.template deserialize<typename MatrixT::Index>());
+
+    MatrixT                                 result(numRows, numCols);
+
+    for (typename MatrixT::Index rowId = 0; rowId < numRows; ++rowId) {
+        for (typename MatrixT::Index colId = 0; colId < numCols; ++colId) {
+            typename MatrixT::Scalar        element(Traits<typename MatrixT::Scalar>::deserialize(ar));
+            result(rowId, colId) = element;
+        }   
+    }
+
+    return result;
+}
+
+} // anonymous namespace
+
+template <typename T>
+struct Traits<Eigen::MatrixX<T>> {

```

Comment:
>Traits [](start = 7, length = 6)

We will need to add additional functionality to make this Trait class compliant with the others. See the list on line 155.